spotify-backup.py

```diff
@@ -45,6 +45,10 @@ def get(self, url, params={}, tries=3):
 	# fetches all pages and joins them, returning in a single list of objects.
 	def list(self, url, params={}):
 		response = self.get(url, params)
+		requests = int(response['total'] / params['limit'])
+		if requests > 10:
+			timeEstimate = int(0.27*requests); # in seconds, 0.28 was the average speed of 130 requests
```

Comment:
I don't think guessing this estimate is a good idea.  
People have all sorts of connections (from 150Kbps to 200Mpbs). This is gonna be spot on for some, and just wrong for others.
Showing the remaining amount of requests should suffice -- it's a clear indicator that things are happening.

OTOH, maybe just [a progress bar](https://tqdm.github.io/) would suffice?

Reply:
The goal was to give a sense of the time it might take (10s of seconds, minutes, ...). I guess I kind of failed that.

I think (just guessing) the time this needs doesn't really directly depend on the speed of the connection. Rather it depends on the ping and the time both the client and server need to respond/send a new request. This time might be more consistent for the average connection.

Does the meaning of this estimate get more clear when adding a (+/- 1000%) after the estimate?

Reply:
A progress bar would look nice. But maybe adding dependencies will make the script harder to use for people who are new to programming but need their playlists.